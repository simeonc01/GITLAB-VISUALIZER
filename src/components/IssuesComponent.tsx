import React, { useContext, useEffect, useState } from "react";
import { Issue } from "../util/types";
import { GitLabContext } from "./GitlabProvider";
import Box from "@mui/material/Box";
import Collapse from "@mui/material/Collapse";
import IconButton from "@mui/material/IconButton";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import Typography from "@mui/material/Typography";
import Paper from "@mui/material/Paper";
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
import { Description } from "@mui/icons-material";

function IssuesComponent() {
  const [issues, setIssues] = useState<Issue[]>([]);

  const context = useContext(GitLabContext);

  useEffect(() => {
    const tempIssues = context.issues;
    if (tempIssues !== null) setIssues(tempIssues);
    console.log(issues);
  }, [context.issues]);

  function createData(
    title: string,
    assignee: string,
    created: Date,
    price: number,
    description: string
  ) {
    return {
      title,
      assignee,
      created,
      price,
      description,
      history: [
        {
          description: "2020-01-02",
          customerId: "11091700",
          amount: 3,
        },
        {
          date: "2020-01-02",
          customerId: "Anonymous",
          amount: 1,
        },
      ],
    };
  }

  const rows: ReturnType<typeof createData>[] = issues.map((issue) => {
    return createData(
      issue.title,
      "Navn",
      issue.created_at,
      3.3,
      issue.description
    );
  });

  function Row(props: { row: ReturnType<typeof createData> }) {
    const { row } = props;
    const [open, setOpen] = React.useState(false);

    return (
      <React.Fragment>
        <TableRow sx={{ "& > *": { borderBottom: "unset" } }}>
          <TableCell>
            <IconButton
              aria-label="expand row"
              size="small"
              onClick={() => setOpen(!open)}
            >
              {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
            </IconButton>
          </TableCell>
          <TableCell component="th" scope="row">
            {row.title}
          </TableCell>
          <TableCell>{row.assignee}</TableCell>
          <TableCell align="right">{row.created.toString()}</TableCell>
        </TableRow>
        <TableRow>
          <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={6}>
            <Collapse in={open} timeout="auto" unmountOnExit>
              <Box sx={{ margin: 1 }}>
                <Typography variant="h6" gutterBottom component="div">
                  Description
                </Typography>
                <Table size="small" aria-label="purchases">
                  <TableBody>
                    <TableRow>
                      <TableCell>{row.description}</TableCell>
                    </TableRow>
                  </TableBody>
                </Table>
              </Box>
            </Collapse>
          </TableCell>
        </TableRow>
      </React.Fragment>
    );
  }

  return (
    <div>
      <h1>Issues Component</h1>
      <TableContainer component={Paper}>
        <Table aria-label="Commits-table">
          <TableHead>
            <TableRow>
              <TableCell />
              <TableCell>Issue</TableCell>
              <TableCell>Author</TableCell>
              <TableCell align="right">Date</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map((row) => (
              <Row key={row.title} row={row} />
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
}

export default IssuesComponent;
