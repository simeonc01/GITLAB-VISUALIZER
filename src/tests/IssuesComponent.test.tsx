import { expect, test } from "@jest/globals";
import "@testing-library/jest-dom/extend-expect";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import IssuesComponent from "../components/IssuesComponent";

afterEach(() => {
  cleanup();
});

test("Should render Issues component", () => {
  //render(<IssuesComponent />);
  expect(true).toBe(true);
});

test("Matches snapshot", () => {
  const tree = renderer.create(<IssuesComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
