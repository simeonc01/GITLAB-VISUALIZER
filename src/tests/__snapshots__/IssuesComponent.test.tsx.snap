// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div>
  <h1>
    Issues Component
  </h1>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="Commits-table"
      className="MuiTable-root css-rqglhn-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          />
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Issue
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Author
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Date created
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            Date closed
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</div>
`;
